### ISSUE TITLE
Enabling audio device for Tiny11

### ISSUE THREAD

**Issue by geckolinux** [dockur/windows#632]
### Operating system

Debian 12

### Description

Hi there, I'm having trouble enabling an audio device in my Tiny11 installation with this project. For reference, I've tried all the suggestions in #108 , #214 , and #22. This is on a Debian 12 host with Pipewire audio (PulseAudio compatibility enabled) and the Cinnamon desktop. I'm connecting to the Docker image over RDP with Remmina, Vinagre, and `xfreerdp` with the same outcome.

- https://github.com/dockur/windows/issues/108#issuecomment-1913140977
```
environment:
  ARGUMENTS: "-device ich9-intel-hda -device hda-output"
```
```
docker logs Tiny11
❯ Starting Windows for Docker v3.12...
❯ For support visit https://github.com/dockur/windows
❯ CPU: Intel Core TM i7 9750H CPU | RAM: 23/32 GB | DISK: 267 GB (btrfs) | HOST: 6.1.0-22-amd64...

❯ Booting Windows using QEMU v8.2.4...
❯ ERROR: qemu-system-x86_64: -device hda-output: no default audio driver available
```

```
environment:
  ARGUMENTS: "-device usb-audio"
```
```
docker logs Tiny11
❯ Starting Windows for Docker v3.12...
❯ For support visit https://github.com/dockur/windows
❯ CPU: Intel Core TM i7 9750H CPU | RAM: 23/32 GB | DISK: 267 GB (btrfs) | HOST: 6.1.0-22-amd64...

❯ Booting Windows using QEMU v8.2.4...
❯ ERROR: qemu-system-x86_64: -device usb-audio: no default audio driver available
```

https://github.com/dockur/windows/issues/108#issuecomment-1913140977
> The reason why I didnt add it by default is that without any audio device I have working sound via Microsoft RDP on Android. But as soon as I add a soundcard like above, for some strange reason I do not have sound in RDP anymore.
> 
> So another solution is to use Microsoft RDP instead of Teamviewer, because it has working sound even without audio device.

So I tried it with no `ARGUMENTS:`

![image](https://github.com/dockur/windows/assets/2692138/7392321a-379c-431c-9778-99ceb1e74fc0)

- https://github.com/dockur/windows/issues/108#issuecomment-2183821037

`ARGUMENTS: "-device ich9-intel-hda -device hda-output -audio alsa"`

![image](https://github.com/dockur/windows/assets/2692138/9b892987-8a22-418e-9e17-838a9806fa8a)

- https://github.com/dockur/windows/issues/214#issuecomment-1975239329

`ARGUMENTS: "-device ich9-intel-hda -device hda-output -audio spice`

```
docker logs Tiny11
Tiny11     | ❯ Starting Windows for Docker v3.12...
Tiny11     | ❯ For support visit https://github.com/dockur/windows
Tiny11     | ❯ CPU: Intel Core TM i7 9750H CPU | RAM: 23/32 GB | DISK: 267 GB (btrfs) | HOST: 6.1.0-22-amd64...
Tiny11     | 
Tiny11     | ❯ Booting Windows using QEMU v8.2.4...
Tiny11     | ❯ ERROR: 
Tiny11 exited with code 15
```

- https://github.com/dockur/windows/issues/22#issuecomment-2041118563

`ARGUMENTS: "-device ich9-intel-hda,addr=1f.1 -audiodev pa,id=snd0,server=unix:/tmp/pa -device hda-output,audiodev=snd0"`

```
docker logs Tiny11
❯ Starting Windows for Docker v3.12...
❯ For support visit https://github.com/dockur/windows
❯ CPU: Intel Core TM i7 9750H CPU | RAM: 23/32 GB | DISK: 267 GB (btrfs) | HOST: 6.1.0-22-amd64...

❯ Booting Windows using QEMU v8.2.4...
❯ ERROR: qemu-system-x86_64: Unknown audio driver `pa'. Perhaps you want to install qemu-system-gui package?
```

- https://discuss.linuxcontainers.org/t/how-to-add-sound-to-an-ubuntu-lxd-vm/14372/11

`ARGUMENTS: "-device intel-hda -device hda-duplex -audio spice"`

```
docker logs Tiny11
Tiny11     | ❯ Starting Windows for Docker v3.12...
Tiny11     | ❯ For support visit https://github.com/dockur/windows
Tiny11     | ❯ CPU: Intel Core TM i7 9750H CPU | RAM: 23/32 GB | DISK: 267 GB (btrfs) | HOST: 6.1.0-22-amd64...
Tiny11     | 
Tiny11     | ❯ Booting Windows using QEMU v8.2.4...
Tiny11     | ❯ ERROR: 
Tiny11 exited with code 15
```

### Docker compose

```
services:
  windows:
    image: dockurr/windows
    container_name: Tiny11
    environment:
      VERSION: "tiny11"
      DISK_FMT: "qcow2"
    volumes:
      - /home/myself/.var/VMs/Tiny11:/storage
      - /home/myself/Desktop:/shared
    devices:
      - /dev/kvm
    cap_add:
      - NET_ADMIN
    ports:
      - 8006:8006
      - 3389:3389/tcp
      - 3389:3389/udp
    privileged: true
    stop_grace_period: 2m

**Comment by geckolinux** [comment]
OK, finally success. First of all I recreated the container with the compose file I posted above that includes `ARGUMENTS: "-device ich9-intel-hda -device hda-output -audio alsa"` , my first attempt didn't use any arguments. I'm not sure if that made a difference or not, but the other important thing I was missing was enabling audio over RDP.

**Method 1:** `xfreerdp /u:username /p:pass /v:localhost /dynamic-resolution /sound /microphone`

For lower audio latency and lower CPU usage this is what I'm using:
`/microphone:format:1,quality:high /sound:format:1,quality:high,latency:5 /network:lan -encryption /cert:ignore /bpp:32 /gfx-h264:AVC444`

**Method 2:** In Remmina:

![image](https://github.com/dockur/windows/assets/2692138/8c9d3419-efff-4257-aad4-e57dee43930b)
